src/RuleSelection/RuleRunner.cs

```diff
@@ -60,7 +60,6 @@ private static void Run(ScanResults results, A11yElement e)
         private static ScanResult ConvertRunResultToScanResult(RunResult runResult)
         {
             if (runResult == null) throw new ArgumentNullException(nameof(runResult));
-            if (runResult.RuleInfo == null) throw new ArgumentException(nameof(runResult.RuleInfo));
```

Comment:
I understand that CreateResult will throw is runResult.RuleInfo is null, but if someone later changed this method to not call CreateResult, then we're going to have NullReferenceExceptions in the code. It may be "belt and suspenders", but I'd lobby for explicitly throwing *in this method* if runResult.RuleInfo is null on input

Reply:
Fixed